addons/google_calendar/models/google_sync.py

```diff
@@ -188,7 +188,11 @@ def _google_delete(self, google_service: GoogleCalendarService, google_id, timeo
 
     @after_commit
     def _google_patch(self, google_service: GoogleCalendarService, google_id, values, timeout=TIMEOUT):
-        with google_calendar_token(self.env.user.sudo()) as token:
+        calendar_user = self.env.user
+        if self.env.user != self.user_id and self.user_id and self.user_id.google_calendar_rtoken:
+            calendar_user = self.user_id
+
+        with google_calendar_token(calendar_user.sudo()) as token:
```

Comment:
You could add a small method helper that retuens the calendar_user and call it in_google_delete too.

Reply:
Thanks for your feedback.
I'll add that function.
I didn't add it to the delete because I wanted to prevent deleting an event of someone else. What is your opinion?

Reply:
In odoo you can always delete events of other users. with google_calendar, we prefers to archive to allow syncing the state. The event are unlinked at the following sync. If you unlink the event of another user you will probably have an error. We are handling them better in v15.0. That's why I think this PR would be better in master. It's far from perfect in 14.0 but I don't want to do more bad than good for a simple "organizer tag".